contracts/Strategy.sol

```diff
@@ -507,16 +494,26 @@ contract Strategy is BaseStrategy, DydxFlashloanBase, ICallee {
         //This part makes sure our withdrawal does not force us into liquidation
         (uint256 depositBalance, uint256 borrowBalance) = getCurrentPosition();
 
-        uint256 AmountNeeded = 0;
-        if(collateralTarget > 0){
-            AmountNeeded = borrowBalance.mul(1e18).div(collateralTarget);
-        }
-        uint256 redeemable = depositBalance.sub(AmountNeeded);
+        uint256 amountNeeded = 0;
+        if(collateralTarget == 0){
+            collateralTarget = 1e15; // 0.001 * 1e18. lower we have issues
+        } 
+        
+        amountNeeded = borrowBalance.mul(1e18).div(collateralTarget);
 
-        if (redeemable < _amount) {
-            cToken.redeemUnderlying(redeemable);
-        } else {
-            cToken.redeemUnderlying(_amount);
+        if(depositBalance >= amountNeeded){
+            uint256 redeemable = depositBalance.sub(amountNeeded);
+
+            if (redeemable < _amount) {
+                cToken.redeemUnderlying(redeemable);
+            
+            } else {
+                cToken.redeemUnderlying(_amount);
+            }
+        }
+       
+        if(collateralTarget == 0 && want.balanceOf(address(this)) > borrowBalance){
```

Comment:
Question:

When the collateralTarget is 0 at this point? In line 498 (https://github.com/Grandthrax/YearnV2-Generic-Lev-Comp-Farm/pull/1/files#diff-3e804d467472e49fe8afea1899d96f26655f3ecd641555a801abcea4f12af598R498) we set it to 1e15 if it is 0. Please CMIIW if that is the case, the condition in the if is never met.

Reply:
Good spot. fixed in https://github.com/Grandthrax/YearnV2-Generic-Lev-Comp-Farm/pull/1/commits/c6bc54ea62184cd4a3deb99b1725bac9d2d4ee87

Reply:
Perfect! Thanks. So we don't need to update the `collateralTarget` in the `_withdrawSome` function now. Right?

Reply:
Nope, we just use a different number temporarily so the math works